File: nodecore-cli/src/main/java/nodecore/cli/serialization/SignedTransactionInfo.java
// VeriBlock NodeCore CLI
// Copyright 2017-2020 Xenios SEZC
// All rights reserved.
// https://www.veriblock.org
// Distributed under the MIT software license, see the accompanying
// file LICENSE or http://www.opensource.org/licenses/mit-license.php.

package nodecore.cli.serialization;

import com.google.gson.annotations.SerializedName;
import nodecore.api.grpc.VeriBlockMessages;
import nodecore.api.grpc.utilities.ByteStringUtility;

public class SignedTransactionInfo {
    public SignedTransactionInfo(final VeriBlockMessages.SignedTransaction signed) {
        signatureIndex = signed.getSignatureIndex();
        signature = ByteStringUtility.byteStringToHex(signed.getSignature());
        publicKey = ByteStringUtility.byteStringToHex(signed.getPublicKey());
        transaction = new TransactionInfo(signed.getTransaction());
    }

    public SignedTransactionInfo(final VeriBlockMessages.SignedTransactionInfo signed) {
        signatureIndex = signed.getSignatureIndex();
        signature = ByteStringUtility.byteStringToHex(signed.getSignature());
        publicKey = ByteStringUtility.byteStringToHex(signed.getPublicKey());
        transaction = new TransactionInfo(signed.getTransaction().getTransaction());
        confirmations = signed.getTransaction().getConfirmations();
        bitcoinConfirmations = signed.getTransaction().getBitcoinConfirmations();
    }

    public String signature;

    @SerializedName("public_key")
    public String publicKey;

    @SerializedName(("signature_index"))
    public long signatureIndex;

    public TransactionInfo transaction;

    @SerializedName("confirmations")
    public int confirmations;

    @SerializedName("bitcoinConfirmations")
    public int bitcoinConfirmations;
}

File: nodecore-cli/src/main/java/nodecore/cli/serialization/SignedTransactionInfo.kt
// VeriBlock NodeCore CLI
// Copyright 2017-2021 Xenios SEZC
// All rights reserved.
// https://www.veriblock.org
// Distributed under the MIT software license, see the accompanying
// file LICENSE or http://www.opensource.org/licenses/mit-license.php.
package nodecore.cli.serialization

import com.google.gson.annotations.SerializedName
import nodecore.api.grpc.RpcSignedTransaction
import nodecore.api.grpc.RpcSignedTransactionInfo
import org.veriblock.sdk.extensions.toHex

class SignedTransactionInfo(
    val signature: String,
    @SerializedName("public_key")
    val publicKey: String?,
    @SerializedName("signature_index")
    val signatureIndex: Long?,
    val transaction: TransactionInfo?,
    @SerializedName("confirmations")
    val confirmations: Int = 0,
    @SerializedName("bitcoinConfirmations")
    val bitcoinConfirmations: Int = 0
)

fun RpcSignedTransaction.toSignedTransactionInfo(): SignedTransactionInfo = SignedTransactionInfo(
    signatureIndex = signatureIndex,
    signature = signature.toHex(),
    publicKey = publicKey.toHex(),
    transaction = TransactionInfo(transaction)
)

fun RpcSignedTransactionInfo.toModel(): SignedTransactionInfo = SignedTransactionInfo(
    signatureIndex = signatureIndex,
    signature = signature.toHex(),
    publicKey = publicKey.toHex(),
    transaction = TransactionInfo(transaction.transaction),
    confirmations = transaction.confirmations,
    bitcoinConfirmations = transaction.bitcoinConfirmations
)
